base/server/upgrade/11.9.0/02-UpdateConfJavaVersion.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright Red Hat, Inc.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+import logging
+import re
+
+import pki
+logger = logging.getLogger(__name__)
+
+
+class UpdateConfJavaVersion(pki.server.upgrade.PKIServerUpgradeScriptlet):
+    def __init__(self):
+        super().__init__()
+        self.message = "Update Conf Java Version"
+
+    def upgrade_instance(self, instance):
+
+        # Getting correct JAVA_HOME version from /usr/share/pki/etc/pki.conf
+        pki_conf_path = instance.SHARE_DIR + "/etc/pki.conf"
+        logger.info("Pulling JAVA_HOME version from %s", pki_conf_path)
+        java_home = self.get_java_version(pki_conf_path)
```

Comment:
The `get_java_version()` can generate an exception stopping the server to start.